src/main/java/org/codelibs/fess/ds/atlassian/api/util/OAuthUtil.java

```diff
@@ -0,0 +1,132 @@
+        final String nonce = generateNonce();
+        final String timestamp = generateTimestamp();
+        final String signature = generateSignature(consumerKey, privateKey, token, verifier, nonce, timestamp, requestMethod, url);
+        final StringBuilder buf = new StringBuilder("OAuth");
+        appendParameter(buf, "oauth_consumer_key", consumerKey);
+        appendParameter(buf, "oauth_nonce", nonce);
+        appendParameter(buf, "oauth_signature", signature);
+        appendParameter(buf, "oauth_signature_method", SIGNATURE_METHOD);
+        appendParameter(buf, "oauth_timestamp", timestamp);
+        appendParameter(buf, "oauth_token", token);
+        appendParameter(buf, "oauth_verifier", verifier);
+        return buf.substring(0, buf.length() - 1);
+    }
+
+    public static void appendParameter(final StringBuilder buf, final String name, final String value) {
+        buf.append(' ').append(UrlUtil.escape(name)).append("=\"").append(UrlUtil.escape(value)).append("\",");
+    }
+
+
+    public static Map<String, String> getQueryMapFromUrl(final URL url) {
+        final String query = url.getQuery();
+        return Arrays.stream(query.split("&")).collect(Collectors.toMap(p -> p.split("=")[0], p -> p.split("=")[1]));
+    }
+
+    public static String generateSignature(final String consumerKey, final PrivateKey privateKey,
+                                           final String token, final String verifier, final String nonce,
+                                           final String timestamp, final String requestMethod, final URL url) {
+
+        final Map<String, String> parameters = new TreeMap<>();
+        parameters.put( "oauth_consumer_key", consumerKey);
+        parameters.put( "oauth_nonce", nonce);
+        parameters.put( "oauth_signature_method", SIGNATURE_METHOD);
+        parameters.put( "oauth_timestamp", timestamp);
+        parameters.put( "oauth_token", token);
+        parameters.put( "oauth_verifier", verifier);
+
+        final Map<String, String> queryMap = getQueryMapFromUrl(url);
+        for (Map.Entry<String, String> fieldEntry : queryMap.entrySet()) {
+            parameters.put(fieldEntry.getKey(), UrlUtil.escape(fieldEntry.getValue()));
+        }
+
+        final String normalizedParameters = UrlUtil.buildQueryParameters(parameters);
+        final String normalizedPath = url.getProtocol() + "://" + url.getAuthority() + url.getPath();
+        final StringBuffer signatureBaseString = new StringBuffer();
+        signatureBaseString.append(UrlUtil.escape(requestMethod)).append('&');
+        signatureBaseString.append(UrlUtil.escape(normalizedPath)).append('&');
+        signatureBaseString.append(UrlUtil.escape(normalizedParameters));
+
+        return computeSignature(privateKey, signatureBaseString.toString());
+    }
+
+    public static String generateNonce() {
+        return Long.toHexString(Math.abs(RANDOM.nextLong()));
```

Comment:
Add debug log before this line.
SecureRandom is blocked if an entropy is empty.